app/main.py

```diff
@@ -1 +1,26 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(self, name: str, health: int = 100) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = False
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, Health:"
+                f" {self.health}, Hidden: {self.hidden}}}")
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    @staticmethod
+    def bite(victim: Herbivore) -> None:
```

Comment:
The `bite` method in the `Carnivore` class is defined to accept only `Herbivore` instances. This is fine as long as the task requirements specify that only `Herbivore` instances can be bitten. If the task allows for more flexibility, consider accepting a more general type, like `Animal`.